tests/phpunit/integration/Core/Modules/REST_Modules_ControllerTest.php

```diff
@@ -0,0 +1,521 @@
+					'slug' => 'optimize',
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEquals( 'invalid_module', $response->get_data()['code'] );
+		$this->assertEquals( 400, $response->get_status() );
+	}
+
+	public function test_check_access_rest_endpoint__success() {
+		$analytics = $this->modules->get_module( 'analytics' );
+		$analytics->get_client()->setHttpClient( new FakeHttpClient() );
+		$analytics->get_settings()->merge(
+			array(
+				'accountID'             => '12345678',
+				'profileID'             => '12345678',
+				'propertyID'            => '987654321',
+				'internalWebPropertyID' => '1234567890',
+			)
+		);
+
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/modules/data/check-access' );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'slug' => 'analytics',
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertArrayIntersection(
+			array(
+				'access' => true,
+			),
+			$response->get_data()
+		);
+		$this->assertEquals( 200, $response->get_status() );
+	}
+
+	public function test_recover_module_rest_endpoint__no_get_method() {
+		$request  = new WP_REST_Request( 'GET', '/' . REST_Routes::REST_ROOT . '/core/modules/data/recover-module' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEquals( 'rest_no_route', $response->get_data()['code'] );
+	}
+
+	public function test_recover_module_rest_endpoint__requires_module_slug() {
+		$request  = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/modules/data/recover-module' );
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEquals( 'invalid_module_slug', $response->get_data()['code'] );
+		$this->assertEquals( 404, $response->get_status() );
+	}
+
+	public function test_recover_module_rest_endpoint__invalid_module_slug() {
+		$request  = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/modules/data/recover-module' );
+		$response = rest_get_server()->dispatch( $request );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'slug' => 'non-existent-module',
+				),
+			)
+		);
+
+		$this->assertEquals( 'invalid_module_slug', $response->get_data()['code'] );
+		$this->assertEquals( 404, $response->get_status() );
+	}
+
+	public function test_recover_module_rest_endpoint__requires_shareable_module() {
+		$request = new WP_REST_Request( 'POST', '/' . REST_Routes::REST_ROOT . '/core/modules/data/recover-module' );
+		$request->set_body_params(
+			array(
+				'data' => array(
+					'slug' => 'search-console',
+				),
+			)
+		);
+		$response = rest_get_server()->dispatch( $request );
+
+		$this->assertEquals( 'module_not_shareable', $response->get_data()['code'] );
+		$this->assertEquals( 404, $response->get_status() );
+	}
+
+	public function test_recover_module_rest_endpoint__requires_recoverable_module() {
+		$this->enable_feature( 'dashboardSharing' );
```

Comment:
Is the `dashboardSharing` relevant for any of these tests? It doesn't seem to be referenced anywhere in the controller.

If it is, let's add a comment as to why. If not, let's remove it.

Reply:
@asvinb this comment may have been missed before. I don't think the feature needs to be enabled for any of the tests here but let me know if I'm mistaken 👍

Reply:
@aaemnnosttv I tried removing those but the tests will fail. I'am thinking because dashboardSharing is still behind a feature flag, that's why we need to have it enabled to test the recover-module endpoint? 🤔

Reply:
@aaemnnosttv This happens because the `Module::is_sharable()` method used [here](https://github.com/google/site-kit-wp/blob/43c436489a48caeead4c33809187b6823fb0aaaf/includes/Core/Modules/REST_Modules_Controller.php#L594-L605) checks if any datapoint of the module has the `shareable` property.

Currently, all shareable modules enforce a check for the `dashboardSharing` feature flag to return a truthy value for this property. Here is an example:

https://github.com/google/site-kit-wp/blob/43c436489a48caeead4c33809187b6823fb0aaaf/includes/Modules/Search_Console.php#L167

Hence, if the feature flag is not enabled in the test case(s), we will not get achieve the desired assertion because the REST request would terminate right at the `Module::is_sharable()` check.

As suggested, I have added comments to explain why it is required.